lib/matplotlib/tests/test_axes.py

```diff
@@ -6554,9 +6554,12 @@ def test_aspect_nonlinear_adjustable_datalim():
 
     ax = fig.add_axes([.1, .1, .8, .8])  # Square.
     ax.plot([.4, .6], [.4, .6])  # Set minpos to keep logit happy.
-    ax.set(xscale="log", xlim=(1, 10),
-           yscale="logit", ylim=(1/11, 1/1001),
+    ax.set(xscale="log", xlim=(1, 100),
+           yscale="logit", ylim=(1 / 101, 1 / 11),
            aspect=1, adjustable="datalim")
     ax.margins(0)
     ax.apply_aspect()
-    assert ax.get_xlim() == pytest.approx(np.array([1/10, 10]) * np.sqrt(10))
+    # Currently the autoscaler chooses to reduce the x-limits by half a decade
+    # on each end, but this may change later.
```

Comment:
I don't know why it would, unless the test itself is changed.  I think the point of confusion is that in this test you are plotting only one point, so apply_aspect is understandably and correctly shrinking an axis instead of expanding one. xlim and ylim pertain to view limits, not data limits.  It looks like I made a big mistake, long ago, in calling the adjustable option "datalim", because it is not actually adjusting data limits--they remain determined by the actual plot calls and Artist additions--but rather, view limits.  Maybe we should change the option name (with deprecation, in a separate PR) to "viewlim" or "viewlimits"?

Reply:
I just removed the comment and let's punt the discussion of what the ideal behavior should be to, well, not this PR :)